Library/Formula/logstash.rb

```diff
@@ -6,17 +6,31 @@ class Logstash < Formula
 
   bottle :unneeded
 
+  head do
+    url "https://github.com/elastic/logstash.git"
+    depends_on :java => "1.8"
+  end
+
   depends_on :java => "1.7+"
 
   def install
-    inreplace %w[bin/logstash], %r{^\. "\$\(cd `dirname \$0`\/\.\.; pwd\)\/bin\/logstash\.lib\.sh\"}, ". #{libexec}/bin/logstash.lib.sh"
+    if build.head?
+      # Build the package from source
+      system "rake", "artifact:tar"
+      # Extract the package to the current directory
+      mkdir "tar"
+      system "tar", "--strip-components=1", "-xf", Dir["build/logstash-*.tar.gz"].first, "-C", "tar"
+      Dir.chdir "tar"
+    end
```

Comment:
@mikemcquaid To produce a packaging of logstash that matches the packaging in the released versions of logstash. Running bootstrap and installing the core plugins includes only a subset of the plugins distributed with the released versions of logstash.